src/main/io/dataEdition.c

```diff
@@ -0,0 +1,1359 @@
+    selectedItem = -1;
+    shouldInitialiseThePage = false;
+	onGoingEditionProcess = false;
+	numberOfSecondaryPagesForThatPage = 0;
+}
+
+static uint8_t firstItemOnLine(uint8_t line)
+{
+    return line * THREE_ITEMS_PER_LINE;
+}
+
+static void highlightSelectedItemOnOledDisplay(void)
+{
+	switch (structureOfPage.editField[selectedItem].color){
+		case PIXEL_MIXED :
+			oledCharFormat = INVERSE_CHAR_FORMAT;
+			i2c_OLED_set_xy(positionInLineOfInvertedChar, editionLine);
+			i2c_OLED_send_char(contentOfPage.lineOfText[editionLine][positionInLineOfInvertedChar]);
+			break;
+		case PIXEL_INVERTED :
+			if (structureOfPage.editField[selectedItem].color == PIXEL_INVERTED) {
+				oledCharFormat = INVERSE_CHAR_FORMAT;
+				for(int i = structureOfPage.editField[selectedItem].leftPosition; i <= structureOfPage.editField[selectedItem].rightPosition; i++){
+					i2c_OLED_set_xy(i, editionLine);
+					i2c_OLED_send_char(contentOfPage.lineOfText[editionLine][i]);
+				}
+			}
+			break;
+	}
+	oledCharFormat = NORMAL_CHAR_FORMAT;
+}
+
+static void completeStringWithSpaces(uint8_t lineIndex)
+{
+    uint8_t length = strlen(contentOfPage.lineOfText[lineIndex]);
+
+    while (length < sizeof(contentOfPage.lineOfText[lineIndex]))
+    	contentOfPage.lineOfText[lineIndex][length++] = 0x20;
+}
+
+
+#if (DATA_EDITION_FULL_VERSION)
+static bool shouldTriggerAlarmNow(void)
+{
+	return ((millis() - lastAlarmSoundTime) >= (uint32_t)(masterConfig.telemetryConfig.hottAlarmSoundInterval * MILLISECONDS_IN_A_SECOND));
+}
+
+static bool isBatteryLevelOK(void)
+{
+	if ((featureConfigured(FEATURE_VBAT)) && (getBatteryState() != BATTERY_OK))
+		return false;
+	return true;
+}
+
+static bool isRssiLevelOK(void)
+{
+	if (((masterConfig.rxConfig.rssi_channel > 0) || (featureConfigured(FEATURE_RSSI_ADC))) && (rssi <= masterConfig.rxConfig.rssi_warningLevel))
+	    return false;
+	return true;
+}
+
+static void checkForBatteryAndRSSIWarningLevels(void)
+{
+    if (isAlarmSoundOn && shouldTriggerAlarmNow()){
+    	lastAlarmSoundTime = millis();
+	}
+}
+
+static uint8_t numberOfActiveDisplayLinesForMotorsPlusOne(void)
+{
+	return (1 + ceil((float)motorCount / (THREE_ITEMS_PER_LINE - 1)));
+}
+
+static void stopAllMotors(void)
+{
+	for (int index = 0; index < motorCount; index++)
+        motor_disarmed[index] = masterConfig.escAndServoConfig.mincommand;
+}
+
+static void stopAllServos(void)
+{
+	for (int index = 0; index < MAX_SUPPORTED_SERVOS; index++)
+		servo[index] = currentProfile->servoConf[index].middle;
+}
+
+static void getReceiverMapping(void)
+{
+    for (int i = 0; i < MAX_MAPPABLE_RX_INPUTS; i++)
+    	tempString[masterConfig.rxConfig.rcmap[i]] = rcChannelLetters[i];
+}
+
+static uint8_t getChannelIndex(void)
+{
+	for (int i = 0; i < MAX_MAPPABLE_RX_INPUTS; i++)
+    	if (tempString[structureOfPage.editField[19].data] == rcChannelLetters[i])
+            return i;
+    return -1;
+}
+
+static bool theRCMappingStringIsValid(void)
+{
+	for (int i = 0; i < 8; i++)
```

Comment:
I don't think unifying these procedures is worth the effort
as 1) it's only a few bytes, 2) the needed output is different and 3) CLI is likely to be removed.